test/kafkaTracking.js

```diff
@@ -115,7 +115,7 @@ describe('test/kafkaTracking.js >', () => {
     let token;
     const redisUrl = process.env.REDIS_URL || '//127.0.0.1:6379';
     const timestamp = Date.now();
-    const receivedTime = Date.now(); 
+    const receivedTime = Date.now();
 
     before(() => {
       conf.pubSubPerspectives = [redisUrl];
```

Comment:
I see that you changed back the tests - the expected client count is the same as it was before now that the clientCount includes all sockets, because the test doesn't include filtering - it would be good to add filtering so you can actually differentiate from the previous behavior.

Same for pubsubstats.js.

Reply:
Good call

Reply:
@ConalCosgrove

Reply:
Not 100% sure on how to implement this @jgraff2 we are testing the filtering in socketIOEmitter, would the correct course of action be to set up a similar set of filters on the socket io server and run the same  `runFilterTests` function?

Reply:
You can use the same approach, but there's no need to duplicate all the test cases here - I'm thinking just one or two basic filters to test that the filtering actually gets applied before setting the client count.

On line 154 below it currently sets up three clients with the same filters. You could add another three clients that have different filters and modify the event objects sent in the test body so they only match one set of filters. Then there will be six clients, but only three of them should be getting the events, and you can verify that the client count is three rather than six.

Reply:
Loos good.